File: Part1_강의자료/Ch06_누적합/문제별코드/16713_GenericQueries/16713.java
import java.util.Scanner;

class Main
{
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int M = sc.nextInt();
        int[] arr = new int[N + 1];
        for (int i = 1; i <= N; i++)
            arr[i] = sc.nextInt();

        int[] acc = new int[N + 1];
        for (int i = 1; i <= N; i++)
            acc[i] = acc[i - 1] ^ arr[i];

        int ans = 0;
        while (M-- > 0) {
            int i = sc.nextInt();
            int j = sc.nextInt();
            ans ^= acc[j] ^ acc[i - 1];
        }
        System.out.println(ans);
    }
}

File: Part1_강의자료/Ch06_누적합/문제별코드/16713_GenericQueries/16713.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, Q;

    scanf("%d %d", &N, &Q);
    vector<int> arr(N + 1);
    for (int i = 1; i <= N; i++)
        scanf("%d", &arr[i]);

    vector<int> acc(N + 1);
    for (int i = 1; i <= N; i++)
        acc[i] = acc[i - 1] ^ arr[i];

    int ans = 0;
    while (Q--)
    {
        int s, e;
        scanf("%d %d", &s, &e);
        ans ^= acc[e] ^ acc[s - 1];
    }
    printf("%d\n", ans);
    return 0;
}
